oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -698,6 +709,17 @@ private void validatePasswordGrantType() {
 			Assert.hasText(this.tokenUri, "tokenUri cannot be empty");
 		}
 
+		private void validateAuthorizationGrantTypes() {
+			for (AuthorizationGrantType authorizationGrantType : authorizationGrantTypes) {
+				if (authorizationGrantType.getValue().equalsIgnoreCase(this.authorizationGrantType.getValue()) &&
+						!authorizationGrantType.equals(this.authorizationGrantType)) {
+					logger.warn(LogMessage.format("AuthorizationGrantType: %s does not match the pre-defined" +
```

Comment:
You can run `./gradlew format` to format these changes per the [contributing guidelines](https://github.com/spring-projects/spring-security/blob/main/CONTRIBUTING.adoc#mind-the-whitespace).

Reply:
Done, also made me realize I was missing a space.